main.nf

```diff
@@ -956,75 +1059,101 @@ process PICARD_METRICS {
+                                                               ch_mpileup_ivar_bcftools
+
+    script:
+    prefix = params.protocol == 'amplicon' ? "${sample}.trim.mkD" : "${sample}.mkD"
+    """
+    samtools mpileup \\
+        --count-orphans \\
+        --no-BAQ \\
+        --max-depth 50000 \\
+        --fasta-ref $fasta \\
+        --min-BQ $params.min_base_qual \\
+        --output ${prefix}.mpileup \\
+        ${bam[0]}
+    """
+}
+
+/*
+ * STEP 5.6.1: Variant calling with VarScan 2
  */
 process VARSCAN2 {
     tag "$sample"
     label 'process_medium'
     publishDir "${params.outdir}/variants/varscan2", mode: params.publish_dir_mode,
         saveAs: { filename ->
-            		      if (filename.endsWith("vcf.gz")) filename
-                      else if (filename.endsWith("vcf.gz.tbi")) filename
-                      else if (filename.endsWith(".log")) "log/$filename"
+                      if (filename.endsWith(".log")) "log/$filename"
                       else if (filename.endsWith(".txt")) "bcftools_stats/$filename"
-                      else params.save_pileup ? filename : null
+                      else filename
                 }
 
     when:
     !params.skip_variants && 'varscan2' in callers
 
     input:
-    tuple val(sample), val(single_end), path(bam) from ch_sort_bam_varscan2
+    tuple val(sample), val(single_end), path(mpileup) from ch_mpileup_varscan2
     path fasta from ch_fasta
 
     output:
-    tuple val(sample), val(single_end), path("*.highfreq.vcf.gz*") into ch_varscan2_highfreq_snpeff,
-                                                                        ch_varscan2_highfreq_consensus
-    tuple val(sample), val(single_end), path("*.lowfreq.vcf.gz*") into ch_varscan2_lowfreq_snpeff
-    path "*.highfreq.bcftools_stats.txt" into ch_varscan2_bcftools_highfreq_mqc
-    path "*.lowfreq.bcftools_stats.txt" into ch_varscan2_bcftools_lowfreq_mqc
-    path "*.highfreq.varscan2.log" into ch_varscan2_log_highfreq_mqc
-    path "*.lowfreq.varscan2.log" into ch_varscan2_log_lowfreq_mqc
-    path "*.pileup"
+    tuple val(sample), val(single_end), path("${prefix}.vcf.gz*") into ch_varscan2_highfreq_consensus,
+                                                                       ch_varscan2_highfreq_snpeff
+    tuple val(sample), val(single_end), path("${sample}.vcf.gz*") into ch_varscan2_lowfreq_snpeff
+    path "${prefix}.bcftools_stats.txt" into ch_varscan2_bcftools_highfreq_mqc
+    path "*.varscan2.log" into ch_varscan2_log_mqc
+    path "${sample}.bcftools_stats.txt"
 
     script:
+    prefix = "${sample}.AF${params.max_allele_freq}"
     """
-    samtools mpileup \\
-        --count-orphans \\
-        --max-depth 20000 \\
-        --min-BQ 0 \\
-        --fasta-ref $fasta \\
-        ${bam[0]} \\
-        > ${sample}.pileup
-
     varscan mpileup2cns \\
-        ${sample}.pileup \\
-        --min-var-freq 0.02 \\
+        $mpileup \\
+        --min-coverage 10 \\
```

Comment:
I think this a really important parameter that should be selected by user, but it's perfect that is setted to 10 by default.

Reply:
Fixed https://github.com/nf-core/viralrecon/pull/93/commits/a495c1c40bd21085c21247ad812e51bcbb90088f